### PR TITLE
[DOCS] Adds latest transform example to transform docs

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds a `latest` type of continuous transform example to the Examples section of the transform docs. The example covers only the necessary API calls like the rest of the examples and does not contain information about how to create the transform via the UI. A separate PR will add screenshots to the examples to cover the Kibana-side of the process. The example also applies a retention policy.

### Preview

[Examples – Finding the last log event from an IP address](https://elasticsearch_70562.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/transform-examples.html#example-last-log)

## NOTE

Delete the retention policy-related parts from the 7.11 backport PR!

### CODE REVIEW

File: docs/reference/transform/examples.asciidoc [hunk 1]
```diff
@@ -336,3 +337,116 @@ This {transform} makes it easier to answer questions such as:
...
+
+[[example-last-log]]
+== Finding the last log event from an IP address
+
+This example uses the web log sample dataset to find the last log from an IP 
+address. Let's use the `latest` type of {transform} which copies the most recent 
+documents from the source index to the destination index. Pick the `clientip` 
+field as the unique key; the data is grouped by this field. Select `@timestamp` 
+as the date field that sorts the data chronologically. The {transform} creates 
+the destination index that contains the latest login date for each client IP.
```
Comment (hendrikmuhs) [inline docs/reference/transform/examples.asciidoc:350]: I think its hard to sell this feature without running in continuous mode.  We also need some reason why: 

With latest you will be able to quickly get the current(latest) state of all your entities, independent of when the last update occurred. On top of latest you can run further summarizations quick and efficient.

File: docs/reference/transform/examples.asciidoc [hunk 2]
```diff
@@ -336,3 +337,116 @@ This {transform} makes it easier to answer questions such as:
...
+    ],
+    "sort": "@timestamp" <2>
+  },
+  "frequency": "1m",
+  "dest": {
+    "index": "last-log-from-clientip"
+  },
+  "settings": {
+    "max_page_search_size": 500
+  }
```
Comment (hendrikmuhs) [inline docs/reference/transform/examples.asciidoc:373]: as said, we need `sync` in this example, running `latest` as batch is technically possible but I think most users want continuous.

**Review by hendrikmuhs — COMMENTED** [review]
I think we have to make a continuous example from this.

I also wonder if we should include `retention_policy` (however, this is limited to `7.12`), so its covered in an example, too.

/CC @przemekwitek

**Comment by szabosteve** [comment]
@hendrikmuhs My latest commit (56acdfe) transformed the example into a continuous transform example that includes retention policy, too. I'll delete the retention policy-related parts from the 7.11 backport PR (take a note about it in the PR description as well).

File: docs/reference/transform/examples.asciidoc [hunk 3]
```diff
@@ -336,3 +337,149 @@ This {transform} makes it easier to answer questions such as:
...
+}
+
+----------------------------------
+// TEST[skip:setup kibana sample data]
+
+<1> Specifies the filed for grouping the data.
```
Comment (lcawl) [inline docs/reference/transform/examples.asciidoc:405]:
```suggestion
<1> Specifies the field for grouping the data.
```

File: docs/reference/transform/examples.asciidoc [hunk 4]
```diff
@@ -336,3 +337,149 @@ This {transform} makes it easier to answer questions such as:
...
+<2> Specifies the date filed that is used for sorting the data.
```
Comment (lcawl) [inline docs/reference/transform/examples.asciidoc:406]:
```suggestion
<2> Specifies the date field that is used for sorting the data.
```

File: docs/reference/transform/examples.asciidoc [hunk 5]
```diff
@@ -336,3 +337,149 @@ This {transform} makes it easier to answer questions such as:
...
+After you create the {transform}, start it:
+
+[source,console]
+----------------------------------
+POST _transform/last-log-from-clientip/_start
+----------------------------------
+// TEST[skip:setup kibana sample data]
+
+
+After the {transform} stopped, search the destination index:
```
Comment (lcawl) [inline docs/reference/transform/examples.asciidoc:424]: I think continuous transforms don't stop. Maybe this is preferrable?:
```suggestion
After the {transform} processes the data, search the destination index:
```

**Review by hendrikmuhs — APPROVED** [review]
LGTM

**Pull request merged**